src/scaffoldmaker/meshtypes/meshtype_3d_lung4.py

```diff
@@ -0,0 +1,804 @@
+            options["Lower lobe base concavity"] = 0.1
+            options["Ventral edge sharpness factor"] = 0.9
+            options["Cardiac curvature"] = 2.0
+        else:
+            options["Lower lobe base concavity"] = 0.0
+            options["Ventral edge sharpness factor"] = 0.0
+            options["Cardiac curvature"] = 0.0
+
+        return options
+
+    @classmethod
+    def getOrderedOptionNames(cls):
+        return [
+            "Left lung",
+            "Right lung",
+            "Number of elements lateral",
+            "Number of elements lower lobe extension",
+            "Number of elements oblique",
+            "Number of transition elements",
+            "Ellipsoid height",
+            "Ellipsoid dorsal-ventral size",
+            "Ellipsoid medial-lateral size",
+            "Left-right lung spacing",
+            "Lower lobe base concavity",
+            "Lower lobe extension",
+            "Ventral edge sharpness factor",
+            "Cardiac curvature",
+            "Refine",
+            "Refine number of elements"
+        ]
+
+    @classmethod
+    def checkOptions(cls, options):
+        dependent_changes = False
+
+        max_transition_count = None
+        for key in [
+            "Number of elements lower lobe extension"
+        ]:
+            if options[key] < 1:
+                options[key] = 1
+        for key in [
+            "Number of elements lateral",
+            "Number of elements oblique"
+        ]:
+            if options[key] < 4:
+                options[key] = 4
+            elif options[key] % 2:
+                options[key] += 1
+            transition_count = (options[key] // 2) - 1
+            if (max_transition_count is None) or (transition_count < max_transition_count):
+                max_transition_count = transition_count
+
+        if options["Number of transition elements"] < 1:
+            options["Number of transition elements"] = 1
+        elif options["Number of transition elements"] > max_transition_count:
+            options["Number of transition elements"] = max_transition_count
+            dependent_changes = True
+
+        for key in [
+            "Ellipsoid height",
+            "Ellipsoid dorsal-ventral size",
+            "Ellipsoid medial-lateral size"
+        ]:
+            if options[key] <= 0.0:
+                options[key] = 1.0
+        for key in [
+            "Lower lobe base concavity",
+            "Lower lobe extension",
+            "Cardiac curvature"
+        ]:
+            if options[key] < 0.0:
+                options[key] = 0.0
+        if options["Lower lobe extension"] == 0.0:
+            options["Number of elements lower lobe extension"] = 0
+            dependent_changes = True
+        depth = options["Ellipsoid dorsal-ventral size"]
+        height = options["Ellipsoid height"]
+        max_extension = 0.99 * magnitude(getEllipsePointAtTrueAngle(depth / 2.0, height / 2.0, math.pi / 3.0))
```

Comment:
returns a long string of numbers 0.4634829428966771. Consider truncating /  rounding to less decimal places?